oauth2/oauth2.go

```diff
@@ -192,9 +192,7 @@ func (c *Client) newAuthenticatedRequest(urlToken string, values url.Values) (*h
 		if err != nil {
 			return nil, err
 		}
-		encodedID := url.QueryEscape(c.creds.ID)
-		encodedSecret := url.QueryEscape(c.creds.Secret)
-		req.SetBasicAuth(encodedID, encodedSecret)
+		req.SetBasicAuth(c.creds.ID, c.creds.Secret)
```

Comment:
This is not OK. You want to revert: https://github.com/coreos/go-oidc/commit/949bc532d603ceb13749b45c8091bddc8579cb07
IMHO this should be fixed in the IDP code - received values should be decoded there.

Also, Keycloak doesn't decode received values properly - https://github.com/keycloak/keycloak/blob/e1af9f133f45bb89a17071b3d8b85fb39c0663cd/core/src/main/java/org/keycloak/util/BasicAuthHelper.java#L52-L61 
If I use a special character in the cliend ID ('@'), then I get the error from Keycloak:
```
unauthorized_client: UNKNOWN_CLIENT: Client was not identified by any client authenticator
```

Reply:
Hm, bummer. 
@jangaraj is correct, so I'm retracting the PR for the time being.